### PR TITLE
refactor(redb-store): Optimization for small file import in redb store

### PR DESCRIPTION
**Pull request by rklaehn**
## Description

Optimization for small file import in redb store.

This batches writes and reads up to some delay and vastly improves performance especially for writes. Linux kernel import goes from minutes to seconds.

## Notes & open questions

~~Currently delete is just a normal operation. I think it should be changed to a toplevel operation because crashing with a delete transaction being not committed would lead to inconsistent state: files would be gone on disk and there in the db.~~

File deletions of any of the "owned" files are now handled outside the transaction. That requires keeping track of deletes and undeletes within a transaction. E.g. if within a single transaction it is determined that hash 0xabcd is to be deleted, it gets marked for deletion, then it gets created again. We must **not** delete the file at the end of the transaction.

## Change checklist

- [x] Self-review.
- [x] Documentation updates if relevant.
- [ ] Tests if relevant.

### CODE REVIEW

**Review by Frando — COMMENTED** [review]
The overall design is very neat, with the categorizing and batched iteration. Some comments inline. I think we can copy this design for the docs store as well, there we also have many very small transactions currently sometimes.

File: iroh-bytes/src/store/file.rs [hunk 1]
```diff
@@ -1429,16 +1483,46 @@ impl Actor {
...
+        let mut msgs = PeekableFlumeReceiver::new(self.msgs.clone());
+        while let Some(msg) = msgs.peek() {
+            if let ActorMessage::Shutdown = msg {
+                break;
+            }
+            match msg.category() {
+                MessageCategory::TopLevel => {
+                    let msg = msgs.recv().expect("just peeked");
+                    if let Err(msg) = self.state.handle_toplevel(&self.db, msg)? {
+                        msgs.push_back(msg).expect("just recv'd");
```
[the remark this answers is not in the document]
Comment (rklaehn) [reply iroh-bytes/src/store/file.rs:1496]: Yeah, this is copypasta from back when this was also a loop.

File: iroh-bytes/src/store/file.rs [hunk 2]
```diff
@@ -1033,15 +1077,23 @@ impl StoreInner {
...
                 } else {
-                    tracing::debug!("copied {} to {}", path.display(), temp_path.display());
+                    let temp_path = self.temp_file_name();
+                    // copy the data, since it is not stable
+                    progress.try_send(ImportProgress::CopyProgress { id, offset: 0 })?;
+                    if reflink_copy::reflink_or_copy(&path, &temp_path)?.is_none() {
+                        tracing::debug!("reflinked {} to {}", path.display(), temp_path.display());
+                    } else {
+                        tracing::debug!("copied {} to {}", path.display(), temp_path.display());
+                    }
```
[the remark this answers is not in the document]
Comment (rklaehn) [reply iroh-bytes/src/store/file.rs:1107]: copy progress for size will be called in finalize_import_sync. added a comment.

**Review by Frando — APPROVED** [review]
My comments are all addressed, so approving this!
If there are corner cases or unexpected situations, we hopefully catch them during dogfooding, so better to get this in now than later IMO.

*Edit: Needs rebase*

**Pull request merged**